src/impl/pooling_direct.jl

```diff
@@ -41,69 +51,80 @@ for name in (:max, :mean)
 
         # Start with the central region
         w_region, h_region, d_region = central_region
-        @inbounds for batch_idx in 1:size(x)[end],
-            c in 1:out_c,
-            d in d_region,
-            h in h_region,
-            w in w_region
-          
-            # Initialize `m` to `0.0`, or `typemin(T)` or whatever.
+
+        @inbounds for batch_idx in 1:size(x, 5), c in 1:out_c
+            for d in d_region
+            pd = project(d, stride_d, pad_d_lo)
+            for h in h_region
+            ph = project(h, stride_h, pad_h_lo)
+            for w in w_region
+            pw = project(w, stride_w, pad_w_lo)
```

Comment:
is this change different from what is already the case on master? It looks similar, but its harder to read.

Reply:
On the master `project(d, stride_d, pad_d_lo)` gets computed every iteration, while here it is computed only when `d` changes.